crates/consensus-logic/src/csm/worker.rs

```diff
@@ -0,0 +1,372 @@
+//! Consensus logic worker task.
+
+// TODO massively refactor this module
+
+use std::{sync::Arc, thread};
+
+use strata_db::{
+    traits::*,
+    types::{CheckpointConfStatus, CheckpointEntry, CheckpointProvingStatus},
+};
+use strata_eectl::engine::ExecEngineCtl;
+use strata_primitives::prelude::*;
+use strata_state::{client_state::ClientState, csm_status::CsmStatus, operation::SyncAction};
+use strata_status::StatusTx;
+use strata_storage::{managers::checkpoint::CheckpointDbManager, L2BlockManager};
+use strata_tasks::ShutdownGuard;
+use tokio::{
+    sync::{broadcast, mpsc},
+    time,
+};
+use tracing::*;
+
+use super::{
+    config::CsmExecConfig,
+    message::{ClientUpdateNotif, CsmMessage, ForkChoiceMessage},
+    state_tracker,
+};
+use crate::{errors::Error, genesis};
```

Comment:
We should be consistent with these imports. Either `super` or `crate`. I prefer `crate`.

Reply:
I prefer `super` for neighboring modules deeper in a hierarchy since ideally we'd slim down the high-level exports and those might not be accessible through `crate::`.